python/pyarrow/tests/test_compute_kernels.py

```diff
@@ -0,0 +1,1237 @@
+        'extract_regex': extract_regex_opts,
+        'find_substring': match_substr_opts,
+        'find_substring_regex': match_substr_opts,
+        'index': idx_opts,
+        'index_in': idx_in_opts,
+        'is_in': idx_in_opts,
+        'match_like': match_substr_opts,
+        'match_substring': match_substr_opts,
+        'match_substring_regex': match_substr_opts,
+        'partition_nth_indices': partition_opts,
+        'replace_substring': replace_substr_opts,
+        'replace_substring_regex': replace_substr_opts,
+        'select_k_unstable': select_k_unstable_opts,
+        'split_pattern': split_pattern_opts,
+        'split_pattern_regex': split_pattern_opts,
+        'strptime': strptime_opts,
+        'utf8_center': pad_opts,
+        'utf8_lpad': pad_opts,
+        'utf8_ltrim': trim_opts,
+        'utf8_replace_slice': replace_slice_opts,
+        'utf8_rpad': pad_opts,
+        'utf8_rtrim': trim_opts,
+        'utf8_trim': trim_opts,
+        'utf8_slice_codeunits': slice_opts
+    }
+
+
+samples_opts = get_sample_opts()
+
+
+class SampleCall:
+
+    def __init__(self, function_name: str, args: List[pa.Array], options,
+                 parameters_map):
+        self.function_name = function_name
+        self.args = args
+        self.options = options
+        self.parameters_map = parameters_map
+
+    def __repr__(self):
+        arg_str = ', '.join([str(arr.type) for arr in self.args])
+        return f'{self.function_name}({arg_str})'
+
+
+def get_sample_empty_calls():
+    for function_name in pc.function_registry().list_functions():
+        if function_name not in function_taxonomy:
+            continue
+        for signature in function_taxonomy[function_name]:
+            for arg_types, _return_type, parameters_map in get_sample_calls(
+                    signature):
+                args = [pa.array([], type=arg_type) for arg_type in arg_types]
+                options_fn = samples_opts.get(function_name, None)
+                options = None
+                if options_fn is not None:
+                    options = options_fn(arg_types)
+                yield SampleCall(function_name, args, options, parameters_map)
+
+
+@pytest.mark.parametrize('function_name',
+                         pc.function_registry().list_functions(), ids=str)
+def test_all_functions_in_taxonomy(function_name):
+    if function_name.startswith('array_'):
+        pytest.xfail(
+            'ARROW-13873: array_* functions should be hidden from python')
+    if function_name.endswith('_meta_binary'):
+        pytest.xfail(
+            'ARROW-13949: *_meta_binary functions should be '
+            'hidden from python')
+    if function_name.startswith('hash_'):
+        pytest.xfail(
+            'ARROW-13943: hash_* functions should be hidden from python')
+    if function_name == 'list_element':
+        pytest.xfail('ARROW-13594: list_element requires a scalar input')
+    assert function_name in function_taxonomy
```

Comment:
Though digging at it now, I think it's just the sheer number of cases that takes a while.

Reply:
I replaced the parameterization with a for loop and the runtime drops down to 2 seconds.  We don't get the nice output showing how many test cases were skipped but I don't know how useful that was at the end of the day.  If you still want it you can run `pytest -s` and it will show a summary at the end of the test.